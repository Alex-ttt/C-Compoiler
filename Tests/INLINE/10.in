inline int abs(int arg){
	if(arg > 0)
		return arg;
	else
		return -arg;		
}

void main(){
	int a, b;
	a = abs(b);
}