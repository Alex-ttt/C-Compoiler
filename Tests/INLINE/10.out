 Symbols Table:
   abs  int function abs(
          arg int
  )
    {
      Statements:
      if

          arg
        >
          0

      then

        {
          Statements:
          return
            arg
--------------------------
        }
      else

        {
          Statements:
          return
            -
              arg
--------------------------
        }
--------------------------
      Symbols Table:
    }
   main  void function main(
  )
    {
      Statements:
      {
        Statements:
          param0
        =
          b
--------------------------
        if

            param0
          >
            0

        then

          {
            Statements:
              inline_temp0
            =
              param0
            goto: BLOCK_LABEL6
--------------------------
          BLOCK_LABEL5:
          }
        else

          {
            Statements:
              inline_temp0
            =
              -
                param0
            goto: BLOCK_LABEL6
--------------------------
          BLOCK_LABEL4:
          }
--------------------------
      BLOCK_LABEL6:
        Symbols Table:
      }
--------------------------
        a
      =
        inline_temp0
--------------------------
      Symbols Table:
              a int
              b int
              inline_temp0 int
    }
